pkg/controller/handlers/toolreference/migrate.go

```diff
@@ -0,0 +1,57 @@
+package toolreference
+
+import (
+	"github.com/obot-platform/nah/pkg/router"
+	v1 "github.com/obot-platform/obot/pkg/storage/apis/obot.obot.ai/v1"
+)
+
+var toolMigrations = map[string]string{
+	"file-summarizer-file-summarizer": "file-summarizer",
+}
+
+func (h *Handler) MigrateToolNames(req router.Request, _ router.Response) error {
+	if len(toolMigrations) == 0 {
+		return nil
+	}
+
+	var tools []string
+
+	switch o := req.Object.(type) {
+	case *v1.Agent:
+		tools = o.Spec.Manifest.Tools
+	case *v1.Workflow:
+		tools = o.Spec.Manifest.Tools
+	case *v1.Thread:
+		tools = o.Spec.Manifest.Tools
+	case *v1.WorkflowStep:
+		tools = o.Spec.Step.Tools
+
+	default:
+		return nil
```

Comment:
Can we add a warning log message here to indicate that the type isn't being migrated?

Reply:
Technicailly this handler doesn't apply to other objects... so this actually will never happen